core/src/calleeFunctions/CurveLpTokenUniv3Callee.ts

```diff
@@ -0,0 +1,53 @@
+import type { CalleeFunctions, CollateralConfig } from '../types';
+import { ethers } from 'ethers';
+import BigNumber from '../bignumber';
+import { getContractAddressByName, getJoinNameByCollateralType } from '../contracts';
+import { convertCrvethToEth, CURVE_COIN_INDEX, CURVE_POOL_ADDRESS } from './helpers/curve';
+import { encodeRoute, convertCollateralToDai } from './helpers/uniswapV3';
+
+export const CHARTER_MANAGER_ADDRESS = '0x8377CD01a5834a6EaD3b7efb482f678f2092b77e';
```

Comment:
Maybe just for me. What exactly is this Charter Manager and why do we need it? I ran into it a couple of times now, but I don't completely understand it.

Reply:
I don't think I can summarise it in a non-maker fashion, since it's still unclear to me as well. But here is the readme https://github.com/makerdao/dss-charter. The callee [suppose to work without it](https://github.com/makerdao/exchange-callees/blob/0e3c637fd762d75ce87e5032425de509e48be179/src/CurveLpTokenUniv3Callee.sol#L133-L135) but it didn't [during tests](https://github.com/sidestream-tech/unified-auctions-ui/pull/156#issuecomment-1098154707), so I had to introduce it 😞

Reply:
For some background on institutational / long term vaults and the `DssCharter` as a vehicle to enable these you can read:
https://forum.makerdao.com/t/mip59-dsscharter/10028
https://forum.makerdao.com/t/institutional-and-long-term-vaults-proposal/9586